backend/common/plugin/spawn.go

```diff
@@ -85,16 +85,17 @@ type Plugin[Client PingableClient] struct {
 //	FTL_WORKING_DIR - the path to a working directory that the plugin can write state to, if required.
 func Spawn[Client PingableClient](
 	ctx context.Context,
+	logger *log2.Logger,
 	name, dir, exe string,
 	makeClient rpc.ClientFactory[Client],
 	options ...Option,
 ) (
 	plugin *Plugin[Client],
-	// "cmdCtx" will be cancelled when the plugin process stops.
+// "cmdCtx" will be cancelled when the plugin process stops.
 	cmdCtx context.Context,
 	err error,
 ) {
-	logger := log2.FromContext(ctx).Sub(name, log2.Default)
+	logger = logger.Sub(name, log2.Default)
```

Comment:
I noticed that `Sub` here doesn't work because I'm technically passing in a different logger than the existing `log.FromContext`. So it will update the deployment logs logger, but not the original one.

Reply:
What does "doesn't work" mean?